src/coreclr/vm/object.cpp

```diff
@@ -1033,6 +1033,41 @@ OBJECTREF::OBJECTREF(const OBJECTREF & objref)
 }
 
 
+//-------------------------------------------------------------
+// VolatileLoadWithoutBarrier constructor
+//-------------------------------------------------------------
+OBJECTREF::OBJECTREF(const OBJECTREF *pObjref, tagVolatileLoadWithoutBarrier tag)
```

Comment:
The tag is there so that this constructor can't accidentally be called. Effectively, it prevents someone from writing...

```
OBJECTREF newObjRef(&oldObjectRef);
```

As that's a very funny construct, that likely doesn't mean what the developer would think it would mean. The tag argument makes it impossible to accidentally use this api. Its not actually used for anything other than that.